src/update.c

```diff
@@ -365,6 +367,37 @@ static int main_update()
 	retries = 0;
 	timeout = 10;
 
+	/* Get a list of all the iterative manifests from the server_manifest that match
+	* the "from" version, also get a list of all the full manifests in the MoM */
+	struct list *list1;
+	struct file *file1;
+
+	list1 = server_manifest->manifests;
+	while (list1) {
```

Comment:
One this that is just a question of preference and I'm not sure if I said that before. But I find so much clearer to loop through a list using for instead of while. So we could replace this:

    list1 = server_manifest->manifests;
    while (list1) {
    list1 = list1->next;

to:

    for (list1 = server_manifest->manifests; list; list = list->next) {

With the advantage that list will be pointing to the current element of the list inside the for block.

Reply:
I like it